### ISSUE TITLE
HTTPS requests not being made.

### ISSUE THREAD

**Issue by JorisBolsens** [robolectric/robolectric#1650]
Hi,
I am trying to test a project that uses volley. 
one of my tests makes sure I am getting proper responses from an API (to ensure it is being used properly).
I have set `Robolectric.getFakeHttpLayer().interceptHttpRequests(false);` 
I am using `final CountDownLatch signal = new CountDownLatch(1);` & `signal.await(30, TimeUnit.SECONDS);` to wait on the response (I have `signal.countDown()` in the response handler).

I am making the request to `https://api.mydomain.com/createAcc` and have a log showing that the request gets added to volley's queue.

However, I do not see the request at all on the server's logs, and am not getting any response which leads me to believe that the request is not actually being made.

My queue is being made with `Volley.newRequestQueue(mContext.getApplicationContext())` and is also being started correctly.

I assume this is an issue with robolectric's mock api, as when I use postman to make the same request, from the same device, the request reaches the server and gets a response just fine.

TL;DR
Trying to make https request in volley, request doesn't reach server

**Comment by JorisBolsens** [comment]
Update, I have been able to get it to work with the following queue code:
```java
public RequestQueue newRequestQueueForTest(final Context context, final OkHttpClient okHttpClient) {
        System.out.println("Creating test queue");
        HttpsTrustManager.allowAllSSL();
        final File cacheDir = new File(context.getCacheDir(), "volley");

        //final Network network = new BasicNetwork(new HttpClientStack(AndroidHttpClient.newInstance("volley")));
        final Network network = new BasicNetwork(new HurlStack());
        final ResponseDelivery responseDelivery = new ExecutorDelivery(Executors.newSingleThreadExecutor());

        final RequestQueue queue =
                new RequestQueue(
                        new DiskBasedCache(cacheDir),
                        network,
                        4,
                        responseDelivery);

        queue.start();

        return queue;
    }
```

However I still get a `connection reset` error which I assume is Volley's problem, not robolectric.
